### ISSUE TITLE
Crashes: docker-sync-stack start

### ISSUE THREAD

**Issue by Vasiliy-Bondarenko** [xeger/docker-compose#34]
From: https://github.com/EugenMayer/docker-sync/issues/49

```
docker-sync start
```
works ok. 
```
docker-sync-stack start 
```
fails.

```
> docker-sync-stack start
/Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:105:in `private': nil is not a symbol (TypeError)
	from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:105:in `<class:Mapper>'
	from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:5:in `<module:Compose>'
	from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:2:in `<top (required)>'
	from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose.rb:10:in `require_relative'
	from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose.rb:10:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/tasks/stack/stack.thor:5:in `<top (required)>'
	from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:22:in `load'
	from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:22:in `block in <top (required)>'
	from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:20:in `each'
	from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:20:in `<top (required)>'
	from /usr/local/bin/docker-sync-stack:23:in `load'
	from /usr/local/bin/docker-sync-stack:23:in `<main>'
```
```
> gem list | grep docker-sync
docker-sync (0.0.13)
```
all yaml files are valid.

```
docker-compose -f docker-compose.yml -f docker-compose-dev.yml up
```
works without problems...

Files:  #[Archive.zip](https://github.com/EugenMayer/docker-sync/files/363227/Archive.zip)

**Comment by xeger** [comment]
Hmm. Some things changed between Ruby 2.0 and 2.1. Try with 2.1 as a
workaround. Easy issue to fix if Eugen would be willing. I am camping this
weekend, away from computers.
On Jul 14, 2016 1:03 AM, "Vasiliy-Bondarenko" <notifications@github.com>
wrote:

From: EugenMayer/docker-sync#49
<https://github.com/EugenMayer/docker-sync/issues/49>

docker-sync start

works ok.

docker-sync-stack start

fails.

> docker-sync-stack start
/Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:105:in
`private': nil is not a symbol (TypeError)
    from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:105:in
`<class:Mapper>'
    from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:5:in
`<module:Compose>'
    from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose/mapper.rb:2:in
`<top (required)>'
    from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose.rb:10:in
`require_relative'
    from /Library/Ruby/Gems/2.0.0/gems/docker-compose-0.8.3/lib/docker/compose.rb:10:in
`<top (required)>'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in
`require'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in
`require'
    from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/tasks/stack/stack.thor:5:in
`<top (required)>'
    from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:22:in
`load'
    from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:22:in
`block in <top (required)>'
    from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:20:in
`each'
    from /Library/Ruby/Gems/2.0.0/gems/docker-sync-0.0.13/bin/docker-sync-stack:20:in
`<top (required)>'
    from /usr/local/bin/docker-sync-stack:23:in `load'
    from /usr/local/bin/docker-sync-stack:23:in `<main>'

> gem list | grep docker-sync
docker-sync (0.0.13)

all yaml files are valid.

docker-compose -f docker-compose.yml -f docker-compose-dev.yml up

works without problems...

Files: #Archive.zip
<https://github.com/EugenMayer/docker-sync/files/363227/Archive.zip>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/xeger/docker-compose/issues/34>, or mute the thread
<https://github.com/notifications/unsubscribe/AAEemCoQy6qMgJ9oc64mDpqpHW0nDqbqks5qVe1sgaJpZM4JMLc3>
.

**Comment by Vasiliy-Bondarenko** [comment]
updated ruby to latest 2.3.0p0
problem fixed for me. thank you.

**Comment by EugenMayer** [comment]
Must be the yaml parser, which ruby version have you been using before ?

**Comment by Vasiliy-Bondarenko** [comment]
2.0
macos default, i beleive.

**Comment by Duske** [comment]
Can confirm this issue with ruby 2.0.0, after upgrading via rvm to 2.3.0p0 it does not occur anymore. But when I stop it with `CTRL-C`, a strange error is reported:

```
Stopping your-simplest-app ...
/Users/dchabrowski/.rvm/gems/ruby-2.3.0/gems/docker-compose-0.8.3/lib/docker/compose/session.rb:201: warning: conflicting chdir during another chdir block
Stopping your-simplest-app ...
Exception in thread Thread-3:
Traceback (most recent call last):
  File "threading.py", line 810, in __bootstrap_inner
  File "threading.py", line 763, in run
  File "compose/cli/log_printer.py", line 190, in watch_events
  File "compose/project.py", line 343, in events
  File "site-packages/docker/client.py", line 251, in _stream_helper
  File "json/__init__.py", line 338, in loads
  File "json/decoder.py", line 369, in decode
ValueError: Extra data: line 2 column 1 - line 3 column 1 (char 634 - 1268)
Stopping your-simplest-app ... done
```

But I guess this is another issue not related to the original one.

**Comment by xeger** [comment]
@Duske, that looks like an error in the actual `docker-compose` tool (which is written in Python); furthermore, it looks like a very weird case where its own requests to Docker's REST API are returning data that it doesn't understand. I would double-check that the version of Docker Compose you are using is compatible with the version of Docker Engine you are using.

**Comment by xeger** [comment]
@Vasiliy-Bondarenko, I have published docker-compose v0.8.4 with Ruby 2.0 support.

**Comment by luikore** [comment]
It can't be installed in Ruby 2.0, since it has a dependency in gemspec which requires ruby version ~> 2.1

**Comment by xeger** [comment]
Thank you @luikore. Published 1.0.1 to remedy this.
